config/Abigail.am

```diff
@@ -12,17 +12,37 @@
 # a mechanism for suppressing harmless warnings.
 #
 
-PHONY += checkabi storeabi
+PHONY += checkabi storeabi checklibabiversion
 
-checkabi:
+checklibabiversion:
+	libabiversion=`abidw -v | $(SED) 's/[^0-9]//g'`; \
+	if test $$libabiversion -lt "200"; then \
+		/bin/echo -e "\n" \
+		    "*** Please use libabigail 2.0.0 version or newer;\n" \
+		    "*** otherwise results are not consistent!\n" \
+		    "(or see https://github.com/openzfs/libabigail-docker )\n"; \
+		exit 1; \
+	fi;
+
+checkabi: checklibabiversion
```

Comment:
`uname -m` will give `amd64` in freebsd, but I've found `TARGET_CPU_X86_64` https://github.com/openzfs/zfs/blob/60ffc1c460e4cdf3c3ca12c8840fd0675a98ed0d/config/always-arch.m4#L37 , what do you think about:

```
if !TARGET_CPU_X86_64
checkabi: allow_only_for_x86_64
storeabi: allow_only_for_x86_64
endif

allow_only_for_x86_64:
        echo '*** ABI definitions provided apply only to x86_64 architecture'; \
        echo '*** Skipping this target and assuming success.'; \
        exit 0

```

, so you don't need to duplicate checks.

Reply:
A good catch! Relying on `uname`'s output is meh.